### PR TITLE
refine bindinfo_test.TestGlobalAndSessionBindingBothExist

### LINKED ISSUE #15880: bindinfo_test: Unstable `TestGlobalAndSessionBindingBothExist`
## Bug Report

The `TestGlobalAndSessionBindingBothExist` in `bindinfo_test` is unstable like:

```
[2020-03-17T01:55:20.255Z] FAIL: bind_test.go:322: testSuite.TestGlobalAndSessionBindingBothExist
[2020-03-17T01:55:20.255Z] 
[2020-03-17T01:55:20.255Z] bind_test.go:400:
[2020-03-17T01:55:20.255Z]     tk.MustIndexLookup("select * from t where a > -1")
[2020-03-17T01:55:20.255Z] /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:203:
[2020-03-17T01:55:20.255Z]     tk.c.Assert(has, check.IsTrue)
[2020-03-17T01:55:20.255Z] ... obtained bool = false
```

Some failed cases can be found here:
- [PR-15876](https://github.com/pingcap/tidb/pull/15876): 
 https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/29695/pipeline/
- [PR-15869](https://github.com/pingcap/tidb/pull/15869): 
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_sqllogic_test_1/detail/tidb_ghpr_sqllogic_test_1/10184/pipeline/98

### PR DESCRIPTION
**Pull request by lawyerphx**
### What problem does this PR solve?

Issue Number: close #15880 

Problem Summary:

### What is changed and how it works?

What's Changed:  bindinfo_test.TestGlobalAndSessionBindingBothExist

How it Works:
* Test the case when global and session binding both exist
* Separate parts of tests to make it clear.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

### CODE REVIEW

**Comment by lawyerphx** [comment]
PTAL ~ @lzmhhh123

File: bindinfo/bind_test.go [hunk 1]
```diff
@@ -341,16 +341,34 @@ func (s *testSuite) TestGlobalAndSessionBindingBothExist(c *C) {
...
 
+	// Test 'drop global binding'
+	c.Assert(tk.HasPlan("SELECT * from t1,t2 where t1.id = t2.id", "MergeJoin"), IsTrue)
+	tk.MustExec("drop global binding for SELECT * from t1,t2 where t1.id = t2.id")
+	c.Assert(tk.HasPlan("SELECT * from t1,t2 where t1.id = t2.id", "HashJoin"), IsTrue)
+
+	// Test the case when global and session binding both exist
+	tk.MustExec("create global binding for SELECT * from t1,t2 where t1.id = t2.id using SELECT  /*+ TIDB_SMJ(t1, t2) */  * from t1,t2 where t1.id = t2.id")
+	c.Assert(tk.HasPlan("SELECT * from t1,t2 where t1.id = t2.id", "MergeJoin"), IsTrue)
+	tk.MustExec("create binding for SELECT * from t1,t2 where t1.id = t2.id using SELECT  /*+ TIDB_SMJ(t1, t2) */  * from t1,t2 where t1.id = t2.id")
```
Comment (lzmhhh123) [inline bindinfo/bind_test.go:364]: We'd better use another join method to distinguish how the session bindings take effort.

Comment (lawyerphx) [reply bindinfo/bind_test.go:364]: done

**Comment by lawyerphx** [comment]
\run-unit-test

**Comment by lawyerphx** [comment]
run-unit-test

**Comment by lawyerphx** [comment]
PTAL~ @eurekaka

File: bindinfo/bind_test.go [hunk 2]
```diff
@@ -341,16 +341,34 @@ func (s *testSuite) TestGlobalAndSessionBindingBothExist(c *C) {
...
+
+	// Test the case when global and session binding both exist
+	tk.MustExec("create global binding for SELECT * from t1,t2 where t1.id = t2.id using SELECT  /*+ TIDB_HJ(t1, t2) */  * from t1,t2 where t1.id = t2.id")
+	c.Assert(tk.HasPlan("SELECT * from t1,t2 where t1.id = t2.id", "HashJoin"), IsTrue)
+	tk.MustExec("create binding for SELECT * from t1,t2 where t1.id = t2.id using SELECT  /*+ TIDB_SMJ(t1, t2) */  * from t1,t2 where t1.id = t2.id")
+	c.Assert(tk.HasPlan("SELECT * from t1,t2 where t1.id = t2.id", "MergeJoin"), IsTrue)
+	tk.MustExec("drop global binding for SELECT * from t1,t2 where t1.id = t2.id")
+	c.Assert(tk.HasPlan("SELECT * from t1,t2 where t1.id = t2.id", "MergeJoin"), IsTrue)
+	tk.MustExec("create global binding for SELECT * from t1,t2 where t1.id = t2.id using SELECT  /*+ TIDB_SMJ(t1, t2) */  * from t1,t2 where t1.id = t2.id")
+	tk.MustExec("drop binding for SELECT * from t1,t2 where t1.id = t2.id")
```
Comment (eurekaka) [inline bindinfo/bind_test.go:369]: Add comments around this line to explain why the following query chooses hash join?

Comment (lawyerphx) [reply bindinfo/bind_test.go:369]: OK

**Pull request merged**